### ISSUE TITLE
How to support chinese select/copy text features

### ISSUE THREAD

**Issue by GoogleCodeExporter** [fleszar1/flexpaper#183]
```
What steps will reproduce the problem?
1.use pdf2swf convert pdf to swf with xpdf 
2.select text
3.copy to notepad

What is the expected output? What do you see instead?

you will see a string 'ffffffff'

how to support chinese select/copy text 
give me a help
```

Original issue reported on code.google.com by `zach0...@qq.com` on 8 Apr 2011 at 2:49

**Comment by GoogleCodeExporter** [comment]
```
I meet the same question,please help chinese men,thank you!
```

Original comment by `yuyang2...@gmail.com` on 22 Apr 2011 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
do you have a sample document? I have seen other chinese documents work
```

Original comment by `erik.eng...@devaldi.com` on 17 Jun 2011 at 2:21

**Comment by GoogleCodeExporter** [comment]
```
I have recognized the same issue with the attached swf:

1. Select / copy Chinese text 巫石 in FlexPaper -> paste to notepad: failed 
(become 'ff')
2. Select / copy Chinese text 巫石 in PDF -> paste to FlexPaper search 
window: success
3. Search '巫石' in FlexPaper: success ('巫石' highlighted)
4. Search '巫' in FlexPaper: failed (no text highlighted)
```

Original comment by `richard...@gmail.com` on 27 Oct 2011 at 7:07

Attachments:
- [LIS-5440046005770898677.pdf](https://storage.googleapis.com/google-code-attachments/flexpaper/issue-183/comment-3/LIS-5440046005770898677.pdf)
- [LIS-5440046005770898677.swf](https://storage.googleapis.com/google-code-attachments/flexpaper/issue-183/comment-3/LIS-5440046005770898677.swf)

**Comment by GoogleCodeExporter** [comment]
```
My pdf2swf parameters used:

-q -f -T 9 -t -s storeallcharacters
```

Original comment by `richard...@gmail.com` on 27 Oct 2011 at 7:08

**Comment by GoogleCodeExporter** [comment]
```
Dear all,

Please kindly help if you have any idea on how to fix the chinese text copy and 
search highlight issues, thanks!!

Richard Shiu
```

Original comment by `richard...@gmail.com` on 31 Oct 2011 at 7:12

**Comment by GoogleCodeExporter** [comment]
```
After debugging the source code downloaded from the SVN site I find that:

1. TextMapUtil.checkUnicodeIntegrity() incorrectly recognized the Chinese 
characters as "invalid" text and convert these characters to "ff"

I have commented line 2276 of Viewer.as:

// _currentlySelectedText = 
TextMapUtil.checkUnicodeIntegrity(_currentlySelectedText,null,_libMC);

And how the Chinese text copying works fine.

2. For the failure to highlight the matched character when search text consists 
of only a single Chinese character:

It turns out that the drawCurrentSelection() of Viewer.as requires at least 2 
characters for the text highlighting to work properly.

==> Follow up question:

Is it possible to change the drawCurrentSelection() implementation so that text 
highlighting also works if only a single character is searched?

Your response is always highly appreciated!!!
```

Original comment by `richard...@gmail.com` on 1 Nov 2011 at 7:54

**Comment by GoogleCodeExporter** [comment]
```
fixed in upcoming build 1.4.7
```

Original comment by `erik.eng...@devaldi.com` on 2 Nov 2011 at 10:49
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Dear Erik,

Thanks so much for the fix!!

Is text highlighting for single character search supported in build 1.4.7 as 
well?
```

Original comment by `richard...@gmail.com` on 3 Nov 2011 at 2:51

**Comment by GoogleCodeExporter** [comment]
```
Dear Erik,

I have noticed that there is an attempt to fix issue #1 in stopSelecting() in 
line 2378-2379 as follows:

if(ResourceManager.getInstance().localeChain[0] != "zh_CN")
    _currentlySelectedText = TextMapUtil.checkUnicodeIntegrity(_currentlySelectedText,null,_libMC);

However, apart from simplified chinese (zh_CN), there are many other languages 
which use multibyte characters (e.g. zh_TW, zh_HK, jp, kr etc).

This languages will pass the "if" condition but 
TextMapUtil.checkUnicodeIntegrity() would not work for them.

Please find a better solution to cope with this situation as there are many 
Asian users who would be using FlexPaper, thank you!!
```

Original comment by `richard...@gmail.com` on 7 Feb 2012 at 10:57

**Comment by GoogleCodeExporter** [comment]
```
I had similar problem.
I am korean.

paste string is 'ff ff fff '

but, I can't Compile Flex library Project that canable download source.

How to Compile Flex Library Project to SWF File.

I want to added source 
'if(ResourceManager.getInstance().localeChain[0] != "zh_CN")'
=> 'if(ResourceManager.getInstance().localeChain[0] != "ko_KR")'

help me .
```

Original comment by `easys0...@gmail.com` on 22 Jan 2014 at 8:31
